modules/monitoring-supported-remote-write-authentication-settings.adoc

```diff
@@ -0,0 +1,314 @@
+<2> The username contained in the specified `Secret` object.
+<3> The password contained in the specified `Secret` object.
+
+.Sample YAML for bearer token authentication using a `Secret` Object
+
+The following shows bearer token settings for a `Secret` object named `rw-bearer-auth` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rw-bearer-auth
+  namespace: openshift-monitoring
+stringData:
+  token: <authentication_token> <1>
+type: Opaque
+----
+<1> The authentication token.
+
+The following shows sample bearer token config map settings that use a `Secret` object named `rw-bearer-auth` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    enableUserWorkload: true
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://authorization.example.com/api/write"
+        authorization:
+          type: Bearer
+          credentials:
+            name: rw-bearer-auth <1>
+            key: token <2>
+----
+<1> The name of the `Secret` object that contains the authentication credentials.
+<2> The authentication token contained in the specified `Secret` object.
+
+.Sample YAML for OAuth 2.0 authentication
+
+The following shows sample OAuth 2.0 settings for a `Secret` object named `oauth2-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: oauth2-credentials
+  namespace: openshift-monitoring
+stringData:
+  id: <oauth2_id> <1>
+  secret: <oauth2_secret> <2>
+  token: <oauth2_authentication_token> <3>
+type: Opaque
+----
+<1> The Oauth 2.0 ID.
+<2> The OAuth 2.0 secret.
+<3> The OAuth 2.0 token.
+
+The following shows an `oauth2` remote write authentication sample configuration that uses a `Secret` object named `oauth2-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://test.example.com/api/write"
+        oauth2:
+          clientId:
+            secret:
+              name: oauth2-credentials <1>
+              key: id <2>
+          clientSecret:
+            name: oauth2-credentials <1>
+            key: secret <2>
+          tokenUrl: https://example.com/oauth2/token <3>
+          scopes: <4>
+          - <scope_1>
+          - <scope_2>
+          endpointParams: <5>
+            param1: <parameter_1>
+            param2: <parameter_2>
+----
+<1> The name of the corresponding `Secret` object. Note that `ClientId` can alternatively refer to a `ConfigMap` object, although `clientSecret` must refer to a `Secret` object.
+<2> The OAuth 2.0 credentials contained in the specified `Secret` object.
```

Comment:
```suggestion
<2> The key in the specified `Secret` object containing the OAuth 2.0 credentials.
```

Reply:
Edited per your suggestion as follows:

`<2> The key that contains the OAuth 2.0 credentials in the specified `Secret` object.
`